### PR TITLE
FIX-#4409: Fix `eval_insert` utility that doesn't actually check results of `insert` function

### LINKED ISSUE #4409: Modin `eval_insert` util function doesn't actually check results of `insert` operation
If parameter `__inplace__=True` is not passed into `eval_general`, then two `None` values will always be compared.

Link: https://github.com/modin-project/modin/blob/0ab04e104a60419c464fc1833a13f7f2b92e7983/modin/pandas/test/dataframe/test_map_metadata.py#L64

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4409 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by anmyachev** [comment]
Blocked by #4408

**Comment by RehanSD** [comment]
I believe this should be unblocked now, right?

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1161,7 +1161,11 @@ def insert(self, loc, column, value, allow_duplicates=False):  # noqa: PR01, D20
                 and not isinstance(value, (pandas.Series, Series))
                 and len(value) != len(self.index)
             ):
-                raise ValueError("Length of values does not match length of index")
+                raise ValueError(
+                    "Length of values ({}) does not match length of index ({})".format(
+                        len(value), len(self.index)
+                    )
+                )
```
Comment (anmyachev) [inline modin/pandas/dataframe.py:1168]: In the process of manual testing, I accidentally found a message mismatch in an exception regarding pandas.

File: modin/pandas/test/dataframe/test_map_metadata.py [hunk 2]
```diff
@@ -71,6 +71,7 @@ def eval_insert(modin_df, pandas_df, **kwargs):
         modin_df,
         pandas_df,
         operation=lambda df, **kwargs: df.insert(**kwargs),
+        __inplace__=True,
```
Comment (anmyachev) [inline modin/pandas/test/dataframe/test_map_metadata.py:74]: [Insert function](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.insert.html?highlight=insert#pandas.DataFrame.insert) doesn't return anything, so we should explicitly use `__inplace__=True` here.

**Review by dchigarev — APPROVED** [review]
good you found it!

**Pull request merged**